### DISCUSSION TITLE
User credentials login page hanged

### DISCUSSION THREAD

**Discussion by EmadKhella** [pnp/powershell#2428] [General]
### What I want to do
I want to loop through all communication sites in my environment to get some details about root web.
I use Connect-pnpOnline to loop on all selected sites using -UsewebLogin 
it is working as expected for sometimes and show the login page and close it because it is using my current saved login details but  after that the login page is hanging for long time with no errors

### PNP powershell version
I am running the latest version of powershell PNP 

### Script:
this is the script I am using 
connect-PnPOnline -url $SiteURL -UseWebLogin
 $siteColl = (Get-PnPTenantSite -Template SITEPAGEPUBLISHING#0) | Select Title,Url,Owner
foreach($site in $siteColl)
{  
    #Get date Created
    Set-PnPTenantSite -Owners $SiteAdminUser -Identity $site.Url 
    Connect-PnPOnline -Url $site.Url -UseWebLogin
    $web=Get-PnPWeb -Includes Created
    $CreatedDate=$web.Created
    $SiteOwners=Get-PnPGroup -AssociatedOwnerGroup |Select Users 
    $userEmails="" 
    foreach($Owner in $SiteOwners.Users)
    {
        $userEmails= $userEmails +$Owner.Email +";"
    }
     Remove-PnPSiteCollectionAdmin -Owners $SiteAdminUser
}  
Disconnect-PnPOnline
![image](https://user-images.githubusercontent.com/54295822/176811781-ee803b54-02e1-4d29-91af-5c3ed3e3d0cb.png)
If I closed the login page I got this error 
![image](https://user-images.githubusercontent.com/54295822/176821237-836a84b7-e8c7-4300-b68f-ec4c913ea1e1.png)

**Comment by KoenZomers** [comment]
@EmadKhella which PnP PowerShell version are you using exactly? Is it a nightly release? I have made some changes after 1.10 that may be causing this. Can you confirm if you're on 1.10 or a later version please?

**Comment by EmadKhella** [comment]
@KoenZomers 
version 1.10.0

**Comment by KoenZomers** [comment]
In that case it cannot be due to the code changes. This is a hard one to troubleshoot as I cannot reproduce it myself. Would it be an option for you to switch to a ClientId with certificate instead to mitigate the issue?

**Comment by EmadKhella** [comment]
@KoenZomers 
I am looping through 600 communication sites. So may be the huge number of sites causing this issue ???

**Comment by veronicageek** [comment]
@EmadKhella - Have you tried Koen's suggestion?

**Comment by EmadKhella** [comment]
@veronicageek It is not applicable in production environment to use ClientID with certificate so I was using WebLogin

**Comment by gautamdsheth** [comment]
Hi looks like the number of sites is the problem. 
Maybe you can reduce the number of sites in the loop and try.
As mentioned in the above comments, the recommendation is to use `-Interactive` or `ClientId+cert` authentication mode.

We wont be fixing this issue as Weblogin is now considered legacy and deprecated.
Am moving this to discussion so that other community members can also chime in and help with their inputs.
